jboss-eap-clustered-vmss-rhel/README.md

```diff
@@ -0,0 +1,336 @@
+
+- Once the deployment starts the resources being deployed will be visible on the deployment page. In the case of any deployment failure, after parameter validation process, a more detailed failure message is available.
+
+- If your deployment fails at the **VMSS Custom Script Extension** resource, a more detailed failure message is available in the VMSS instance log file. Please refer to the next section for further troubleshooting.
+
+#### Troubleshooting EAP deployment extension
+
+This quickstart template uses VMSS Custom Script Extension to deploy and configure JBoss EAP with a sample application. Your deployment can fail at this stage due to several reasons such as:
+
+- Invalid RHSM or EAP entitlement
+- Invalid JBoss EAP or RHEL OS entitlement Pool ID
+
+Follow the steps below to troubleshoot this further:
+
+1. Log into the provisioned VMSS instance through SSH as mentioned in the 'Validation section'.
+
+2. Switch to root user
+
+    `sudo su -`
+
+3. Enter the VMSS instance admin password if prompted.
+
+4. Change directory to logging directory
+
+    `cd /var/lib/waagent/custom-script/download/1`
+
+5. Review the logs in jbosseap.install.log log file.
+
+    `more jbosseap.install.log`
+
+This log file will have details that include deployment failure reason and possible solutions. If your deployment failed due to RHSM account or entitlements, please refer to 'Subscriptions and Costs' section to complete the prerequisites and try again. Note that after your Azure subscription receives access to Red Hat Gold Images, you can locate them in the Azure portal. Go to **Create a Resource** > **See all**. At the top of the page, you'll see that **You have private offers available** bar, click on *View private offers* link to view your private offers.
+
+![alt text](images/private-offer.png)
+
+Your Private Offer will be tagged with *Private* as seen below.
+
+![alt text](images/rhel-byos.png)
+
+Please refer to [Using the Azure Custom Script Extension Version 2 with Linux VMs](https://docs.microsoft.com/azure/virtual-machines/extensions/custom-script-linux) for more details on troubleshooting VM custom script extensions.
+
+#### Cleanup
+
+If your deployment fails due to any reason (like the ones mentioned above), make sure you delete the whole resource group so that you are not charged for any of the resources deployed so far before failure occurred. In case of successful deployment, follow the same process of deleting the resource group created by the template and related resources (if created in other resource groups) to optimize Azure cost. Once your resources are deleted make sure you log into your RHSM account and remove the system you registered for the deployment of the template, so that you do not hit the "Maximum Enabled Entitlement Quantity" (if you have set any). You can remove the registered system under the *Systems* section in the Red Hat portal.

```

Comment:
if I have other resources in the resourceGroup, I probably don't want to delete those

Reply:
Added a note about the same